.travis.yml

```diff
@@ -37,46 +37,58 @@ jobs:
       # the bootstrap above is older historically; this way of doing it is newer
       # and also simpler. we should aim to reduce/eliminate the duplication.
       - stage: build
-        name: bootstrap and publishLocal
+        name: build, publishLocal, build again
         if: type = pull_request OR repo != scala/scala
         script:
           - set -e
-          - sbt -warn setupPublishCore generateBuildCharacterPropertiesFile headerCheck publishLocal
+          - sbt setupPublishCore generateBuildCharacterPropertiesFile headerCheck publishLocal
           - STARR=$(sed -n 's/^maven\.version\.number=//p' buildcharacter.properties) && echo $STARR
-          - sbt -Dstarr.version=$STARR -warn setupValidateTest compile
-          # as per discussion on #9332, caching the new STARR only goes halfway.
-          # we should also cache the products of the bootstrapped build. currently all three stages
-          # (build, testAll1, testAll2) end up recompiling core projects like `library` and `compiler`.
-          # fixing that will take some care to satisfy sbt/zinc.
+          - sbt -Dstarr.version=$STARR setupValidateTest compile
         workspaces:
           create:
-            name: published-local
+            name: bootstrapped
             paths:
-              - "$PWD/buildcharacter.properties"
+              # so new STARR will be available
+              - "buildcharacter.properties"
               - "$HOME/.ivy2/local/org.scala-lang"
+              # so build products built using new STARR are kept
+              - "target"
+              - "project/target"
+              - "project/project/target"
+              - "project/project/project/target"
+              - "dist"
+              - "build"
 
-      - stage: testAll
-        name: testAll1
+      - stage: test
```

Comment:
Can't change this without changing `stages` above.  And changing this to `test` re-enables the default test job (outside of `jobs.include` matrix) which isn't what we want.  You can change it, but to anything but "test".

Reply:
Huh.  The issue I was describing doesn't seem to be present, so I guess if you re-define `stages` so it _doesn't_ include `test` then you don't get the default test.  And then it appends all the stages ad-hoc'ly defined in jobs.include.  So we could probably just null out the stages key with something like `stages: []` and comment how it's removing the default and letting jobs.include define stages.

The question is what's the order of the stages then?  Order of definition?  Alphabetic?  By ruby object hashcode? 🤷

Reply:
I'm not inclined to look into deeper into these imponderables at the moment :-)